core/src/test/scala/higherkindness/mu/rpc/srcgen/AvroSrcGenTests.scala

```diff
@@ -37,6 +43,25 @@ class AvroSrcGenTests extends AnyWordSpec with Matchers with OneInstancePerTest
         forAll { scenario: Scenario => test(scenario) }
       }
     }
+
+    "return a non-empty list of errors instead of generating code from an invalid IDL file" in {
+      val response = {
+        AvroSrcGenerator(
+          List(BigDecimalAvroMarshallers),
+          ScalaBigDecimalTaggedGen,
+          NoCompressionGen,
+          UseIdiomaticEndpoints.trueV
+        ).generateFrom(
+          Source.fromInputStream(getClass.getResourceAsStream("/avro/Invalid.avdl")).mkString,
+          Avro
+        )
+      }
+
+      val expectedResponse =
+        "Some((foo/bar/MyGreeterService.scala,Invalid(NonEmptyList(RPC method response parameter has non-record response type 'STRING'))))"
+
+      assert(response.toString == expectedResponse)
```

Comment:
Would it be possible to check the type instead of the string representation?
```suggestion
      response shouldBe Some(("foo/bar/MyGreeterService.scala", Validated.invalidNel("RPC method response parameter has non-record response type 'STRING'")))
```